src/main/java/com/google/cloud/tools/gradle/appengine/core/ManagedSdkTask.java

```diff
@@ -0,0 +1,53 @@
+package com.google.cloud.tools.gradle.appengine.core;
+
+import com.google.common.base.Strings;
+import java.io.File;
+import org.gradle.api.DefaultTask;
+import org.gradle.api.tasks.TaskAction;
+
+public class ManagedSdkTask extends DefaultTask {
+
+  private ToolsExtension toolsExtension;
+
+  public void setToolsConfig(ToolsExtension toolsExtension) {
+    this.toolsExtension = toolsExtension;
+  }
+
+  /** Task entrypoint : Download/update/verify Cloud SDK installation. */
+  @TaskAction
+  public void managedSdkAction() {
+    getProject().getLogger().lifecycle("Running managedSdkTask.");
+
+    getProject().getLogger().lifecycle("Cloud SDK Version: " + toolsExtension.getCloudSdkVersion());
+    getProject()
+        .getLogger()
+        .lifecycle("Download Cloud SDK: " + toolsExtension.getDownloadCloudSdk());
+    getProject().getLogger().lifecycle("Cloud SDK Home: " + toolsExtension.getCloudSdkHome());
+
+    String sdkVersion = toolsExtension.getCloudSdkVersion();
+    File sdkHome = toolsExtension.getCloudSdkHome();
+    boolean download = toolsExtension.getDownloadCloudSdk();
+
+    if (download) {
```

Comment:
A good way to write this skeleton may be to write an interface that is used by `ManagedSdkTask` in these `TODOs` so that the full code here is written out, but the implementation of the interface is not.

For example, we could have a class like:

```java
class Orchestra {
  private final Conductor conductor;
  void perform() {
    conductor.prepare();
    conductor.conduct();
    conductor.stop();
  }
}
```

Where `Conductor` is just an interface.
When `Conductor` is implemented, it could either be implemented with a separate implementation class, or just be converted to a class itself.

The benefit of this approach is that:

1) There is no unfilled skeleton code that might be accidentally used. (Having an unimplemented interface ensures that the code would not run.)
2) The code is testable - the interface can be mocked (with like Mockito) to run test-specific code so that the logic can be tested. Later, when the interface is implemented, you'll only have to write a test for that interface.